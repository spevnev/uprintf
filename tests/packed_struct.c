#pragma pack(push, 1)
#include "struct.c"
#pragma pack(pop)
