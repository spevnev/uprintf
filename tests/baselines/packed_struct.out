Struct containing all types: {
    uint8_t u8 = 255
    short unsigned int u16 = 65535
    unsigned int u32 = 4294967295
    long long unsigned int u64 = 18446744073709551615
    i8_t i8 = -128
    short int i16 = -32768
    int32_t i32 = -2147483648
    long int i64 = -9223372036854775807
    signed char sch = -99
    unsigned char uch = 99 ('c')
    float f32 = 0.123000
    long_float f64 = -0.321000
    void *void_ptr = POINTER
    int *int_ptr = POINTER (5)
    float *null_ptr = NULL
    const char *str = POINTER ("str string")
    char *const const_str = POINTER ("const_str string")
    _Bool b = true
    void() void_fun = POINTER (void void_fun())
    int(char, float *, long unsigned int) other_fun = POINTER (int other_fun(char a, float *b, long unsigned int c))
}
